### DISCUSSION TITLE
Call to undefined function Statamic\Imaging\simplexml_load_file()

### DISCUSSION THREAD

**Discussion by earthwytch** [statamic/cms#8633] [Troubleshooting] [answered]
In both my local setup (a docker container) and our production server (a docker container running as an Azure App Service) I am randomly getting this error when I try various asset actions in the control panel. Usually the action (e.g. deleting) does occur successfully but I can't see this until later when it rights itself.

```
production.ERROR: Call to undefined function Statamic\Imaging\simplexml_load_file() {"userId":"a307cf86-c614-4ac5-b366-b0f15e25551a","exception":"[object] (Error(code: 0): Call to undefined function Statamic\\Imaging\\simplexml_load_file() at /var/www/localhost/htdocs/vendor/statamic/cms/src/Imaging/Attributes.php:51)
```

Most recently this appeared when simply trying to view the assets in the control panel, but more often appears when trying to delete an asset. It tends to go away when rebooting.

I have two containers - "system" and "media". The error occurs intermittently on both containers and in both environments.

The media container is mounted under /var/www/localhost to allow for an Azure storage drive to be mounted there. In development it is mounted in the same spot as a Docker volume.

```
        'media' => [
            'driver' => 'local',
            'root' => '/var/www/localhost/media',
            'url' => env('APP_URL').'/media',
            'visibility' => 'public',
            'throw' => false,
        ],

        # is persisted in version control by default
        'system' => [
            'driver' => 'local',
            'root' => public_path('system'),
            'url' => '/system',
            'visibility' => 'public',
            'throw' => false,
        ],

    'links' => [
        public_path('storage') => storage_path('app/public'),
        public_path('media') => '/var/www/localhost/media',
    ],
```

Doe anyone have a suggestion as to what might be causing this?

**Comment by jasonvarga** [comment, marked as answer]
You don't have the php module simplexml installed. Maybe you can add this in your docker config.

**Reply by earthwytch** [reply]
Awesome! Thank you - that was easy ⭐
